notebooks/features/cognitive_services/GeospatialServices - Flooding Risk.ipynb

```diff
@@ -0,0 +1,190 @@
+    "\n",
+    "from synapse.ml.geospatial import *\n",
+    "\n",
+    "def extract_location_fields(df):\n",
+    "    # Use this function to select only lat/lon columns into the dataframe\n",
+    "    return df.select(\n",
+    "        col(\"ObjectID\"),\n",
+    "        col(\"SquareFoot\"),\n",
+    "        col(\"AppraisedLandValue\"),\n",
+    "        col(\"AppraisedImprovementValue\"),\n",
+    "        col(\"PresentUse\"),\n",
+    "        col(\"FullAddress\"),\n",
+    "        col(\"output.response.results\").getItem(0).getField(\"position\").getField(\"lat\"),\n",
+    "        col(\"output.response.results\").getItem(0).getField(\"position\").getField(\"lon\")\n",
+    "    )\n",
+    "    \n",
+    "\n",
+    "# Azure Maps geocoder to enhance the dataframe with location data\n",
+    "geocoder = (AddressGeocoder()\n",
+    "    .setSubscriptionKey(azureMapsKey)\n",
+    "    .setAddressCol(\"FullAddress\")\n",
+    "    .setOutputCol(\"output\"))\n",
+    "\n",
+    "# Set up a fixed mini batch transformer to geocode addresses\n",
+    "batched_dataframe = geocoder.transform(FixedMiniBatchTransformer().setBatchSize(10).transform(subset_data.coalesce(1)))\n",
+    "geocoded_addresses = extract_location_fields(FlattenBatch().transform(batched_dataframe)).collect()\n",
+    "\n",
+    "# Display the results\n",
+    "geocoded_addresses.show(5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that we have geocoded the addresses, we can now use the `CheckPointInPolygon` function to check if a property is in a flood zone or not.\n",
+    "\n",
+    "### Setup Check Point In Polygon "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def extract_point_in_polygon_result_fields(df):\n",
+    "    # Use this function to select only lat/lon columns into the dataframe\n",
+    "    return df.select(\n",
+    "        col(\"ObjectID\"),\n",
+    "        col(\"SquareFoot\"),\n",
+    "        col(\"AppraisedLandValue\"),\n",
+    "        col(\"AppraisedImprovementValue\"),\n",
+    "        col(\"PresentUse\"),\n",
+    "        col(\"FullAddress\"),\n",
+    "        col(lat),\n",
+    "        col(lon),\n",
+    "        col(\"output.result.pointInPolygons\"),\n",
+    "        col(\"output.result.intersectingGeometries\")\n",
+    "    )\n",
+    "\n",
+    "\n",
+    "check_point_in_polygon = (CheckPointInPolygon()\n",
+    "    .setSubscriptionKey(azureMapsKey)\n",
+    "    .setGeography(atlas_geo_prefix)\n",
+    "    .setUserDataIdentifier(user_data_id)\n",
+    "    .setLatitudeCol(\"lat\")\n",
+    "    .setLongitudeCol(\"lon\")\n",
+    "    .setOutputCol(\"output\"))\n",
+    "\n",
+    "\n",
+    "flood_plain_addresses = extract_point_in_polygon_result_fields(check_point_in_polygon.transform(geocoded_addresses)).collect()\n",
+    "\n",
+    "# Display the results\n",
+    "flood_plain_addresses.show(5)"
```

Comment:
change to display(...) for nicer results

Reply:
done